sync/sync_head.go

```diff
@@ -239,6 +230,7 @@ func (s *Syncer[H]) verifyBifurcating(ctx context.Context, subjHead, networkHead
 
 		// candidate was validated properly, update subjHead.
 		subjHead = candidateHeader
+		s.setSubjectiveHead(ctx, subjHead)
```

Comment:
How we now collected valid headers along the bifurcation.

Reply:
This change actually allows to refactor bifurcation to be fully recursive with twice as less code, but I am too lazy to do this

Reply:
Feel an issue for that?

Reply:
However...I don't see a reason for that.